Jackson 2.21+ compatibility: InvalidDefinitionException due to conflicting creators in ReplaceAWTGetPeerMethod and ReplaceComSunAWTUtilitiesMethods

## What version of OpenRewrite are you using?

 - rewrite-migrate-java 3.29.0
- jackson-bom 2.21.1

 ## What is the smallest, simplest way to reproduce the problem?

  Upgrade jackson-bom to 2.21.x and run any test that loads a recipe including Java8toJava11,

## What did you see instead?

  RecipeSerializer.write() fails during recipe tree serialization with an InvalidDefinitionException.

  ReplaceAWTGetPeerMethod and ReplaceComSunAWTUtilitiesMethods both use @AllArgsConstructor (Lombok) alongside an explicit no-arg constructor annotated with @JsonCreator. Lombok's @AllArgsConstructor generates a constructor with @ConstructorProperties,
  which Jackson treats as a property-based creator — conflicting with the @JsonCreator no-arg constructor.

  - This was exposed by https://github.com/FasterXML/jackson-databind/issues/5045 (fixed in 2.21.0 via https://github.com/FasterXML/jackson-databind/pull/5301), which changed creator precedence resolution so that conflicting property-based creators are
  now correctly rejected.

## What is the full stack trace of any errors you encountered?

```
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Conflicting property-based creators: already had explicit creator [constructor for `org.openrewrite.java.migrate.ReplaceAWTGetPeerMethod` (0 args), annotations: {interface com.fasterxml.jackson.annotation.JsonCreator=@com.fasterxml.jackson.annotation.JsonCreator(mode=DEFAULT)}, encountered another: [constructor for `org.openrewrite.java.migrate.ReplaceAWTGetPeerMethod` (2 args), annotations: {interface java.beans.ConstructorProperties=@java.beans.ConstructorProperties({"getPeerMethodPattern", "lightweightPeerFQCN"})} (through reference chain: org.openrewrite.config.DeclarativeRecipe["recipeList"]->java.util.ArrayList[1]->org.openrewrite.config.DeclarativeRecipe["recipeList"]->java.util.ArrayList[1]->org.openrewrite.config.DeclarativeRecipe["recipeList"]->java.util.ArrayList[9]->org.openrewrite.config.DeclarativeRecipe["recipeList"]->java.util.ArrayList[0]->org.openrewrite.config.DeclarativeRecipe["uninitializedRecipes"]->java.util.ArrayList[33])
	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1359)
	at com.fasterxml.jackson.databind.SerializerProvider._createAndCacheUntypedSerializer(SerializerProvider.java:1506)
	at com.fasterxml.jackson.databind.SerializerProvider.findContentValueSerializer(SerializerProvider.java:824)
	at com.fasterxml.jackson.databind.ser.impl.PropertySerializerMap.findAndAddSecondarySerializer(PropertySerializerMap.java:90)
	at com.fasterxml.jackson.databind.ser.std.AsArraySerializerBase._findAndAddDynamic(AsArraySerializerBase.java:313)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serializeContents(CollectionSerializer.java:146)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:111)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:25)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:760)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeWithType(BeanSerializerBase.java:643)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serializeContents(CollectionSerializer.java:153)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:111)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:25)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:760)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeWithType(BeanSerializerBase.java:643)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serializeContents(CollectionSerializer.java:153)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:111)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:25)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:760)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeWithType(BeanSerializerBase.java:643)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serializeContents(CollectionSerializer.java:153)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:111)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:25)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:760)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeWithType(BeanSerializerBase.java:643)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serializeContents(CollectionSerializer.java:153)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:111)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serialize(CollectionSerializer.java:25)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:760)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeWithType(BeanSerializerBase.java:643)
	at com.fasterxml.jackson.databind.ser.impl.TypeWrappedSerializer.serialize(TypeWrappedSerializer.java:32)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:503)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:342)
	at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4938)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsBytes(ObjectMapper.java:4171)
	at org.openrewrite.RecipeSerializer.write(RecipeSerializer.java:77)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:177)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:135)
	at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:130)
```

--- a/src/main/java/org/openrewrite/java/migrate/ReplaceAWTGetPeerMethod.java
+++ b/src/main/java/org/openrewrite/java/migrate/ReplaceAWTGetPeerMethod.java
@@ -16,7 +16,6 @@
 package org.openrewrite.java.migrate;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
-import lombok.AllArgsConstructor;
 import lombok.EqualsAndHashCode;
 import lombok.Value;
 import org.jspecify.annotations.Nullable;
@@ -29,7 +28,6 @@ import org.openrewrite.java.tree.J;
 import org.openrewrite.java.tree.TypeUtils;
 import org.openrewrite.java.tree.TypedTree;
 
-@AllArgsConstructor
 @EqualsAndHashCode(callSuper = false)
 @Value
 public class ReplaceAWTGetPeerMethod extends Recipe {
@@ -47,9 +45,13 @@ public class ReplaceAWTGetPeerMethod extends Recipe {
     String lightweightPeerFQCN;
 
     @JsonCreator
-    public ReplaceAWTGetPeerMethod() {
-        getPeerMethodPattern = "java.awt.* getPeer()";
-        lightweightPeerFQCN = "java.awt.peer.LightweightPeer";
+    public ReplaceAWTGetPeerMethod(
+            @Nullable String getPeerMethodPattern,
+            @Nullable String lightweightPeerFQCN) {
+        this.getPeerMethodPattern = getPeerMethodPattern == null ?
+                "java.awt.* getPeer()" : getPeerMethodPattern;
+        this.lightweightPeerFQCN = lightweightPeerFQCN == null ?
+                "java.awt.peer.LightweightPeer" : lightweightPeerFQCN;
     }
 
     String displayName = "Replace AWT `getPeer()` method";

--- a/src/main/java/org/openrewrite/java/migrate/ReplaceComSunAWTUtilitiesMethods.java
+++ b/src/main/java/org/openrewrite/java/migrate/ReplaceComSunAWTUtilitiesMethods.java
@@ -16,9 +16,9 @@
 package org.openrewrite.java.migrate;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
-import lombok.AllArgsConstructor;
 import lombok.EqualsAndHashCode;
 import lombok.Value;
+import org.jspecify.annotations.Nullable;
 import org.openrewrite.ExecutionContext;
 import org.openrewrite.Option;
 import org.openrewrite.Recipe;
@@ -28,7 +28,6 @@ import org.openrewrite.java.JavaVisitor;
 import org.openrewrite.java.MethodMatcher;
 import org.openrewrite.java.tree.J;
 
-@AllArgsConstructor
 @EqualsAndHashCode(callSuper = false)
 @Value
 public class ReplaceComSunAWTUtilitiesMethods extends Recipe {
@@ -77,14 +76,28 @@ public class ReplaceComSunAWTUtilitiesMethods extends Recipe {
     String setComponentMixingCutoutShapePattern;
 
     @JsonCreator
-    public ReplaceComSunAWTUtilitiesMethods() {
-        getAWTIsWindowsTranslucencyPattern = "com.sun.awt.AWTUtilities isTranslucencySupported(com.sun.awt.AWTUtilities.Translucency)";
-        getWindowOpacityPattern = "com.sun.awt.AWTUtilities getWindowOpacity(java.awt.Window)";
-        getWindowShapePattern = "com.sun.awt.AWTUtilities getWindowShape(java.awt.Window)";
-        isWindowOpaquePattern = "com.sun.awt.AWTUtilities isWindowOpaque(java.awt.Window)";
-        isTranslucencyCapablePattern = "com.sun.awt.AWTUtilities isTranslucencyCapable(java.awt.GraphicsConfiguration)";
-        setComponentMixingCutoutShapePattern = "com.sun.awt.AWTUtilities setComponentMixingCutoutShape(java.awt.Component,java.awt.Shape)";
-        setWindowOpacityPattern = "com.sun.awt.AWTUtilities setWindowOpacity(java.awt.Window, float)";
+    public ReplaceComSunAWTUtilitiesMethods(
+            @Nullable String getAWTIsWindowsTranslucencyPattern,
+            @Nullable String isWindowOpaquePattern,
+            @Nullable String isTranslucencyCapablePattern,
+            @Nullable String setWindowOpacityPattern,
+            @Nullable String getWindowOpacityPattern,
+            @Nullable String getWindowShapePattern,
+            @Nullable String setComponentMixingCutoutShapePattern) {
+        this.getAWTIsWindowsTranslucencyPattern = getAWTIsWindowsTranslucencyPattern == null ?
+                "com.sun.awt.AWTUtilities isTranslucencySupported(com.sun.awt.AWTUtilities.Translucency)" : getAWTIsWindowsTranslucencyPattern;
+        this.isWindowOpaquePattern = isWindowOpaquePattern == null ?
+                "com.sun.awt.AWTUtilities isWindowOpaque(java.awt.Window)" : isWindowOpaquePattern;
+        this.isTranslucencyCapablePattern = isTranslucencyCapablePattern == null ?
+                "com.sun.awt.AWTUtilities isTranslucencyCapable(java.awt.GraphicsConfiguration)" : isTranslucencyCapablePattern;
+        this.setWindowOpacityPattern = setWindowOpacityPattern == null ?
+                "com.sun.awt.AWTUtilities setWindowOpacity(java.awt.Window, float)" : setWindowOpacityPattern;
+        this.getWindowOpacityPattern = getWindowOpacityPattern == null ?
+                "com.sun.awt.AWTUtilities getWindowOpacity(java.awt.Window)" : getWindowOpacityPattern;
+        this.getWindowShapePattern = getWindowShapePattern == null ?
+                "com.sun.awt.AWTUtilities getWindowShape(java.awt.Window)" : getWindowShapePattern;
+        this.setComponentMixingCutoutShapePattern = setComponentMixingCutoutShapePattern == null ?
+                "com.sun.awt.AWTUtilities setComponentMixingCutoutShape(java.awt.Component,java.awt.Shape)" : setComponentMixingCutoutShapePattern;
     }
 
     String displayName = "Replace `com.sun.awt.AWTUtilities` static method invocations";

